### PR TITLE
Add 2024 Russian Substituted Holidays

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

Add 2024 Russian Substituted Holidays.

This PR supersedes and resolves #1760 (@summicron502 's)  - with all issues resolved and all suggestions implemented.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

### CODE REVIEW

File: holidays/countries/russia.py [hunk 1]
```diff
@@ -113,11 +113,25 @@ class RUS(Russia):
...
-            (MAY, 8, tr("День Победы")),
+            (FEB, 24, JAN, 1),
+            (MAY, 8, JAN, 8),
+        ),
+        # Substituted Holidays 2024
+        # src: https://www.consultant.ru/document/cons_doc_LAW_455140/
+        2024: (
+            (APR, 29, APR, 27),
+            (APR, 30, FEB, 11),
+            (MAY, 10, JUN, 1),
```
Comment (KJhellico) [inline holidays/countries/russia.py:133]:
```suggestion
            (APR, 30, NOV, 2),
            (MAY, 10, JAN, 6),
```
As in source.

File: holidays/countries/russia.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 from gettext import gettext as tr
 
-from holidays.calendars.gregorian import JAN, FEB, MAY
+from holidays.calendars.gregorian import JAN, FEB, APR, MAY, JUN, DEC
```
Comment (KJhellico) [inline holidays/countries/russia.py:15]:
```suggestion
from holidays.calendars.gregorian import JAN, FEB, APR, MAY, NOV, DEC
```

**Review by KJhellico — COMMENTED** [review]
Although I have an ambivalent feeling about Russia and its holidays, I'm going to offer a correction.

File: holidays/countries/russia.py (hunk 2 above)
Comment (PPsyrius) [reply holidays/countries/russia.py:15]: Good catch, fixed.

**Review by KJhellico — APPROVED** [review]
LGTM.

**Pull request merged**